### ISSUE TITLE
Qt5.12.6 on Windows introduces dependency on Wtsapi32.lib

### ISSUE THREAD

**Issue by sipsorcery** [bitcoin/bitcoin#17749]
Testing with Bitcoin Core msvc builds against Qt5.12.6 has revealed a new library dependency on `Wstapi32.lib` which is part of the [Remote Desktop Services](https://docs.microsoft.com/en-us/windows/win32/api/_termserv/index).

There are only two undefined symbols when `Wstapi32.lib` is not set as a dependency:

  - [WTSQuerySessionInformationW](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsquerysessioninformationw)

  - [WTSFreeMemory](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsfreememory) 

These two calls look safe. Despite that the linking of a remote desktop library to Bitcoin Core is not ideal and has already raised concerns.

Ideally a way will be found to set a Qt configuration parameter so the library is not required. I have posted a [question](https://forum.qt.io/topic/109777/configure-option-for-qt5-12-6-to-avoid-dependency-on-wtsapi32-lib) on the Qt forum to that effect.

The Qt build configuration options below DO NOT work. They come from disabling all features and directories that don't look related to Bitcoin Core and are also from looking at #17736 

````
..\configure -debug-and-release -opensource -platform win32-msvc -opengl desktop -no-shared -no-static-runtime -mp -c++std c++11 -confirm-license -ltcg -static -make libs -make tools  -optimized-tools -pch -qt-libpng -qt-pcre -qt-harfbuzz -system-zlib -no-compile-examples -no-angle -no-cups -no-dbus -no-egl -no-eglfs -no-freetype -no-gif -no-glib -no-gtk -no-icu -no-ico -no-iconv -no-kms -no-linuxfb -no-libjpeg -no-libproxy -no-libudev -no-mtdev -no-opengles3 -no-openssl -no-openvg -no-reduce-relocations -no-sctp -no-sql-db2 -no-sql-ibase -no-sql-oci -no-sql-tds -no-sql-mysql -no-sql-odbc -no-sql-psql -no-sql-sqlite -no-sql-sqlite2 -no-sqlite -no-system-proxies -no-use-gold-linker -nomake examples -nomake tests -no-feature-bearermanagement -no-feature-colordialog -no-feature-commandlineparser -no-feature-concurrent -no-feature-dial -no-feature-dtls -no-feature-ftp -no-feature-http -no-feature-image_heuristic_mask -no-feature-keysequenceedit -no-feature-lcdnumber -no-feature-networkdiskcache -no-feature-networkproxy -no-feature-pdf -no-feature-printdialog -no-feature-printer -no-feature-printpreviewdialog -no-feature-printpreviewwidget  -no-feature-sessionmanager -no-feature-socks5 -no-feature-sqlmodel -no-feature-statemachine -no-feature-syntaxhighlighter -no-feature-textedit -no-feature-texthtmlparser -no-feature-textodfwriter -no-feature-topleveldomain -no-feature-udpsocket -no-feature-undocommand -no-feature-undogroup -no-feature-undostack -no-feature-undoview -no-feature-vnc -no-feature-whatsthis -no-feature-wizard -no-feature-xml -skip qt3d -skip qtactiveqt -skip qtandroidextras -skip qtcanvas3d -skip qtcharts -skip qtconnectivity -skip qtdatavis3d -skip qtdeclarative -skip qtdoc -skip qtgamepad -skip qtgraphicaleffects -skip qtimageformats -skip qtlocation -skip qtmacextras -skip qtmultimedia -skip qtnetworkauth -skip qtpurchasing -skip qtquickcontrols -skip qtquickcontrols2 -skip qtscript -skip qtscxml -skip qtsensors -skip qtserialbus -skip qtserialport -skip qtspeech -skip qtvirtualkeyboard -skip qtwayland -skip qtwebchannel -skip qtwebengine -skip qtwebglplugin -skip qtwebsockets -skip qtwebview -skip qtx11extras -skip qtxmlpatterns -IC:\tools\vcpkg\installed\x64-windows-static\include -LC:\tools\vcpkg\installed\x64-windows-static\lib -prefix C:\Qt5.12.6_x64_static_vs2019
````

A substantial number (20 to 30) of trial and error iterations later a set of successful configure options is below. Unfortunately this still results in a build that requires Wtsapi32.lib.

````
..\configure -debug-and-release -opensource -platform win32-msvc -opengl desktop -no-shared -no-static-runtime -mp -c++std c++11 -confirm-license -ltcg -static -make libs -make tools  -optimized-tools -pch -qt-libpng -qt-pcre -qt-harfbuzz -system-zlib -no-compile-examples -no-angle -no-cups -no-dbus -no-egl -no-eglfs -no-freetype -no-gif -no-glib -no-gtk -no-icu -no-ico -no-iconv -no-kms -no-linuxfb -no-libjpeg -no-libproxy -no-libudev -no-mtdev -no-opengles3 -no-openssl -no-openvg -no-reduce-relocations -no-sctp -no-sql-db2 -no-sql-ibase -no-sql-oci -no-sql-tds -no-sql-mysql -no-sql-odbc -no-sql-psql -no-sql-sqlite -no-sql-sqlite2 -no-sqlite -no-system-proxies -no-use-gold-linker -nomake examples -nomake tests -no-feature-bearermanagement -no-feature-concurrent -no-feature-dial -no-feature-dtls -no-feature-ftp -no-feature-http -no-feature-lcdnumber -no-feature-networkdiskcache -no-feature-networkproxy  -no-feature-sessionmanager -no-feature-socks5 -no-feature-sqlmodel -no-feature-topleveldomain -no-feature-udpsocket -no-feature-vnc -skip qt3d -skip qtactiveqt -skip qtandroidextras -skip qtcanvas3d -skip qtcharts -skip qtconnectivity -skip qtdatavis3d -skip qtdeclarative -skip qtdoc -skip qtgamepad -skip qtgraphicaleffects -skip qtimageformats -skip qtlocation -skip qtmacextras -skip qtmultimedia -skip qtnetworkauth -skip qtpurchasing -skip qtscript -skip qtscxml -skip qtsensors -skip qtserialbus -skip qtserialport -skip qtspeech -skip qtvirtualkeyboard -skip qtwayland -skip qtwebchannel -skip qtwebengine -skip qtwebglplugin -skip qtwebsockets -skip qtwebview -skip qtx11extras -IC:\tools\vcpkg\installed\x64-windows-static\include -LC:\tools\vcpkg\installed\x64-windows-static\lib -prefix C:\Qt5.12.6_x64_static_vs2019
````

**Comment by laanwj** [comment]
Looks like the calls were added in `qwindowscontext.cpp` in qt/qtbase@ecfa33e751115919ddb47a6f641a2b1842c96788 .

**Comment by sipsorcery** [comment]
I got a reply on the Qt forum stating that it's not possible to avoid the `Wstapi32.lib`. The reason the dependency was added was to solve an [intermittent clipboard bug](https://codereview.qt-project.org/c/qt/qtbase/+/233795).

IMHO I don't think this would warrant a patch of Qt given the Bitcoin Core GUI can also use the Windows clipboard and thus be affected by the bug.

Will close this issue and wait until the > Qt5.12 update is proposed for Bitcoin Core before any further action.
